lib/ansible/modules/network/dcnm/dcnm_vrf.py

```diff
@@ -37,237 +39,1042 @@
+            return have_create, have_attach, have_deploy
+
+        if not resp['DATA']:
+            return have_create, have_attach, have_deploy
+
+        for vrf in resp['DATA']:
+            all_vrfs += vrf['vrfName'] + ','
+
+        path = '/rest/top-down/fabrics/{}/vrfs/attachments?vrf-names={}'.format(self.fabric, all_vrfs[:-1])
+
+        all_vrf_attach = dcnm_send(self.module, method, path)
+
+        for vrf in resp['DATA']:
+            t_conf = dict()
+            t_conf.update({'vrfSegmentId': vrf['vrfId']})
+            t_conf.update({'vrfName': vrf['vrfName']})
+
+            vrf.update({'vrfTemplateConfig': json.dumps(t_conf)})
+            del vrf['vrfStatus']
+            have_create.append(vrf)
+
+        all_vrfs = ''
+
+        for vrf_attach in all_vrf_attach['DATA']:
+            if not vrf_attach.get('lanAttachList'):
+                continue
+            attach_list = vrf_attach['lanAttachList']
+            dep_vrf = ''
+            for attach in attach_list:
+                attach_state = False if attach['lanAttachState'] == "NA" else True
+                deploy = attach['isLanAttached']
+                if bool(deploy) and (attach['lanAttachState'] == "OUT-OF-SYNC" or
+                                     attach['lanAttachState'] == "PENDING"):
+                    deploy = False
+
+                if bool(deploy):
+                    dep_vrf = attach['vrfName']
+
+                sn = attach['switchSerialNo']
+                vlan = attach['vlanId']
+                del attach['vlanId']
+                del attach['switchSerialNo']
+                del attach['switchName']
+                del attach['switchRole']
+                del attach['ipAddress']
+                del attach['lanAttachState']
+                del attach['isLanAttached']
+                del attach['vrfId']
+                del attach['fabricName']
+
+                attach.update({'fabric': self.fabric})
+                attach.update({'vlan': vlan})
+                attach.update({'serialNumber': sn})
+                attach.update({'deployment': deploy})
+                attach.update({'extensionValues': ""})
+                attach.update({'instanceValues': ""})
+                attach.update({'freeformConfig': ""})
+                attach.update({'isAttached': attach_state})
+
+            if dep_vrf:
+                all_vrfs += dep_vrf + ","
+
+        have_attach = all_vrf_attach['DATA']
+
+        if all_vrfs:
+            have_deploy.update({'vrfNames': all_vrfs[:-1]})
+
+        # logit(json.dumps(have_create, indent=4, sort_keys=True))
+        # logit(json.dumps(have_attach, indent=4, sort_keys=True))
+        # logit(json.dumps(have_deploy, indent=4, sort_keys=True))
+
+        self.have_create = have_create
+        self.have_attach = have_attach
+        self.have_deploy = have_deploy
+
+
+    def get_want(self):
+        want_create = list()
+        want_attach = list()
+        want_deploy = dict()
```

Comment:
Visually, I really like the `var = dict()`  or `var = list()` syntax but I remember reading an article a while back about performance implications of choosing `dict()` vs `{}` for creating a new dict and `list()` vs `[]` for a new list.

I think based on this we should use `{}` and `[]`

https://stackoverflow.com/questions/17097985/dict-vs-in-python-which-is-better/17098017

Reply:
hmm... thanks for pointing that out, one disadvantage of modern programming languages is availability of many syntactically different ways to achieve the same.

Reply:
Agreed.  I used to use `list()`, `dict()` because IMO they provide some clarity... until I ran across the performance discussions.